gateway/gateway.go

```diff
@@ -473,5 +472,14 @@ func (g *gateway) Close(ctx context.Context) error {
 func (g *gateway) Readings(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error) {
 	g.readingsMu.Lock()
 	defer g.readingsMu.Unlock()
+
+	// no readings available yet
+	if len(g.lastReadings) == 0 {
+		// Tell the collector not to capture the empty data.
+		if extra[data.FromDMString] == true {
```

Comment:
wondering if we need to have an `ok` check but based on tests it seems like we don't